### PR TITLE
feat(edit-page): Edit Contentlet Dialog Language Support #27297

### LINKED ISSUE #27297: Edit Page: Edit Contentlet Dialog Language Support
### User Story
As a dotCMS user, I want the edit contentlet dialog to open in the language I am currently editing the page in, so that I can easily add content in the correct language.

```[tasklist]
### Acceptance Criteria
- [ ] When editing a page in X language, opening the edit contentlet dialog should display the dialog in X.
- [ ] When editing a page in X language, opening the add contentlet dialog should display the dialog in X.
- [ ] The language of the edit contentlet dialog should be synchronized with the language of the page being edited
```

### Proposed Objective
Static content: "Core Features"

### Proposed Priority
Static content: "Priority 3 - Average"

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.:
Static Content: "N/A"

### Assumptions & Initiation Needs
- The user is editing a page and wants to add content in the same language.
- The language settings of the page and the user's preferences are correctly configured.

### Quality Assurance Notes & Workarounds
N/A

### Sub-Tasks & Estimates:
- Implement language detection in the edit contentlet dialog (2 hours)
- Synchronize the language of the dialog with the

### PR DESCRIPTION
**Pull request by hmoreras**
### Proposed Changes
* Edit content dialog set lang filter based on page current lang. 

### Screenshots

https://github.com/dotCMS/core/assets/31667212/29a14919-5671-47db-97bb-3ca4de4df60f

### CODE REVIEW

**Review by oidacra — APPROVED** [review]

**Review by manuelrojas — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/dot-edit-content.component.ts [hunk 1]
```diff
@@ -674,4 +676,16 @@ browse from the page internal links
...
+     * Replaces the value of "_content_lang" in the given string with a new value.
+     * @param {string} inputString - The original string containing the "_content_lang" parameter.
+     * @returns {string} - The updated string with the replaced "_content_lang" value.
+     */
+    private setCurrentContentLang(inputString: string): string {
+        const regex = /(_content_lang=)(\d+)(?=&|$)/;
+        const replacement = `$1${this.pageLanguageId}`;
+
+        return inputString.replace(regex, replacement);
+    }
```
Comment (rjvelazco) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/dot-edit-content.component.ts:690]: Would it be safer to utilize the URL API and replace the query parameters with a URL?

```ts
    private setCurrentContentLang(url: string): string {
          const newUrl = new URL(url);
          newUrl.searchParams.set('_content_lang', this.pageLanguageId);
          return newUrl.toString()
    }
```

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/dot-edit-content.component.ts [hunk 2]
```diff
@@ -410,6 +410,8 @@ browse from the page internal links
                 .getActionUrl($event.data.contentType.variable)
                 .pipe(take(1))
                 .subscribe((url) => {
+                    //TODO: check if the getActionUrl can accept a lang as param.
```
Comment (rjvelazco) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/dot-edit-content.component.ts:413]: Is needed?

**Review by rjvelazco — DISMISSED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/dot-edit-content.component.ts (hunk 1 above)
Comment (fmontes) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/dot-edit-content.component.ts:690]: Safer.

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/dot-edit-content.component.ts (hunk 2 above)
Comment (hmoreras) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/dot-edit-content.component.ts]: no ! removed.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/dot-edit-content.component.ts (hunk 1 above)
Comment (hmoreras) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/dot-edit-content.component.ts]: done.

**Review by fmontes — APPROVED** [review]

**Pull request merged**